dashoffloadmanager/dpuinfoprovider.cpp

```diff
@@ -0,0 +1,89 @@
+#include "dpuinfoprovider.h"
+
+#include "dbconnector.h"
+#include "table.h"
+#include "logger.h"
+#include "tokenize.h"
+
+#include <boost/algorithm/string/join.hpp>
+#include <nlohmann/json.hpp>
+
+using namespace std;
+using namespace swss;
+using json = nlohmann::json;
+
+bool getDpuInfo(vector<DpuInfo> &info)
+{
+    DBConnector db("CONFIG_DB", 0);
+    auto bridge_table = Table(&db, CFG_MID_PLANE_BRIDGE_TABLE_NAME);
+    auto dpus_table = Table(&db, CFG_DPUS_TABLE_NAME);
+    auto dhcp_table = Table(&db, CFG_DHCP_SERVER_IPV4_TABLE_NAME);
+
+    string bridge;
+    bridge_table.hget("GLOBAL", "bridge", bridge);
+    if (bridge.empty())
+    {
+        SWSS_LOG_ERROR("Failed to get brdige info from %s", CFG_MID_PLANE_BRIDGE_TABLE_NAME);
+        return false;
+    }
+
+    vector<string> dpus;
+    dpus_table.getKeys(dpus);
+    if (dpus.empty())
+    {
+        SWSS_LOG_ERROR("Failed get DPU list from %s table", CFG_DPUS_TABLE_NAME);
+        return false;
+    }
+
+    sort(dpus.begin(), dpus.end());
+
+    for (uint32_t dpuId = 0; dpuId < dpus.size(); dpuId++)
+    {
+        string dpuMidplane;
+        dpus_table.hget(dpus[dpuId], "midplane_interface", dpuMidplane);
```

Comment:
do you know where is this schema is defined?

Reply:
It's defined in the "Smart Switch IP address assignment" HLD:
https://github.com/sonic-net/SONiC/blob/master/doc/smart-switch/ip-address-assigment/smart-switch-ip-address-assignment.md#npu-platformjson